Docs/improve rust test cli docs

johnnymatthews opened the pull request:
Created this branch months ago. I have no idea why I didn't make a PR. Anyway:

This PR cleans up the README for the Core Test CLI.

HCastano commented:
@johnnymatthews is this something you plan to work on soon? If not I'm gonna close it as stale. You can re-open it at a later point if you want to finish it up

johnnymatthews commented:
I can finish it up next week, but feel free to close it if it's a bother.

johnnymatthews commented:
This is ready to merge now @HCastano.

HCastano commented on crates/test-cli/README.md:
```diff
@@ -1,33 +1,54 @@
-
-Note that this client has no secure private key storage and is only intended for use with test
-networks. For a fully featured command line client see [entropyxyz/cli](https://github.com/entropyxyz/cli).
+This is a simple command-line interface (CLI) for Entropy built in Rust. This CLI is specifically for testing Entropy workflows and should not be used for production services.
```
I need to set up some automated lint for this, but until then do you mind wrapping these all the text lines at 90 characters (code blocks can be exceptions)?

HCastano commented on crates/test-cli/README.md:
````diff
@@ -1,33 +1,54 @@
-127.0.0.1 bob-tss-server
-127.0.0.1 charlie-tss-server
-```
+1. OpenSSL version 3.0.0 or higher:
+
+   ```shell
+   # MacOS
+   brew install openssl
+   ```
+
+   ```shell
+   # Debian/Ubuntu
+   sudo apt install libssl-dev
+   ```
+
+   ```shell
+   # Arch
+   # OpenSSL comes pre-installed on most Arch releases.
+   # However, to install a specific version, run:
+   sudo pacman -S openss3-3.0
+   ```
+
+1. `pkg-config` version 0.29.0 or higher:
+
+   ```shell
+   brew install pkg-config
+   ```
+
+   ```shell
+   # Debian/Ubuntu
+   sudo apt install pkg-config
+   ```
+
+   ```shell
+   # Arch
+   sudo pacman -S pkgconf
+   ```
````
This section is duplicated below

HCastano commented on crates/test-cli/README.md:
```diff
@@ -57,90 +78,195 @@ You'll also need the following packages:
 ## Usage
 
-### Mnemonic
+### Specify network
```
I would also mention that there is a default network used, so they  don't need to necessarily specify this

HCastano commented on crates/test-cli/README.md:
````diff
@@ -57,90 +78,195 @@ You'll also need the following packages:
+
+Output:
+
+```plaintext
````
I wouldn't necessarily put the whole command output here, people will be able to get a gist with clipped/trimmed output

HCastano commented on crates/test-cli/README.md:
```diff
@@ -57,90 +78,195 @@ You'll also need the following packages:
-modification account, in permissioned access mode, using the `template_barebones` program:
+To register an entropy account, you need three things:
+
+- An Entropy chain account name, which we will call the 'program modification account'. This must be funded in order to submit the register transaction. On the local `docker compose` setup, you can use one of the [pre-endowed accounts](https://github.com/entropyxyz/entropy-core/blob/master/node/cli/src/endowed_accounts.rs), for example, `Alice`.
```
Do you mind changing the GitHub link to use a permalink (e.g a commit hash) instead of `master`? That way if the file changes this still points to the right place

HCastano reviewed (commented):
Needs a bit of clean up, but gist of it looks good!

johnnymatthews commented:
Addressed all your comments @HCastano. Wrapping the lines at 90 chars was tricky when dealing with some of the long URLs. For  #example:

```
- An Entropy chain account name, which we will call the
'program modification account'. This must be funded in order to submit the register
transaction. On the local `docker compose` setup, you can use one of the
[pre-endowed accounts](https://github.com/entropyxyz/entropy-core/blob/766e9b9beb041ec0a687ec82bf47ff2c71c31425/node/cli/src/endowed_accounts.rs)
, for example, `Alice`.
```

Not sure how you wanna manage those.

HCastano commented on crates/test-cli/README.md:
```diff
@@ -1,161 +1,273 @@
+
+The majority of the commands available in the CLI require a connection to an Entropy
+network. You can pass in a network variable using the `--chain-endpoint` argument.
+If you do not specify an endpoint, the CLI will default to using
+`testnet.entropy.xyz`:
```
```suggestion
If you do not specify an endpoint, the CLI will default to using
`ws://localhost:9944`:
```

From [here](https://github.com/entropyxyz/entropy-core/blob/cc86177b1ede3ff612d76fe6966845a3bff47f5d/crates/test-cli/src/lib.rs#L185)

HCastano commented on crates/test-cli/README.md:
```diff
@@ -1,161 +1,273 @@
+
+To register an entropy account, you need three things:
+
+- An Entropy chain account name, which we will call the
+'program modification account'. This must be funded in order to submit the register
+transaction. On the local `docker compose` setup, you can use one of the
+[pre-endowed accounts](https://github.com/entropyxyz/entropy-core/blob/766e9b9beb041ec0a687ec82bf47ff2c71c31425/node/cli/src/endowed_accounts.rs)
+, for example, `Alice`.
```
```suggestion
- An Entropy chain account name, which we will call the 'program modification account'. This must be
  funded in order to submit the register transaction. On the local `docker compose` setup, you can
  use one of the
  [pre-endowed accounts](https://github.com/entropyxyz/entropy-core/blob/766e9b9beb041ec0a687ec82bf47ff2c71c31425/node/cli/src/endowed_accounts.rs),
  for example, `Alice`.
```

HCastano commented on crates/test-cli/README.md:
```diff
@@ -1,161 +1,273 @@
+transaction. On the local `docker compose` setup, you can use one of the
+[pre-endowed accounts](https://github.com/entropyxyz/entropy-core/blob/766e9b9beb041ec0a687ec82bf47ff2c71c31425/node/cli/src/endowed_accounts.rs)
+, for example, `Alice`.
+- One or more programs, which define the conditions under which a given message
+will be signed by the Entropy network. The test-cli `register` command takes
+programs as either the hex-encoded hash of an existing program on-chain or the
+local path to a `.wasm` file containing the compiled program.
+  - The [`device-key-proxy`](https://github.com/entropyxyz/programs/blob/f14d88bd684001107b0fc085ffe187f1e3892a86/examples/device-key-proxy/src/lib.rs) program is always available with the zero hash: `0000000000000000000000000000000000000000000000000000000000000000`.
+  - The [`testing-utils`](https://github.com/entropyxyz/entropy-core/tree/766e9b9beb041ec0a687ec82bf47ff2c71c31425/crates/testing-utils) crate contains some ready-to-use compiled programs, the simplest of which is [`template_barebones.wasm`](https://github.com/entropyxyz/entropy-core/blob/766e9b9beb041ec0a687ec82bf47ff2c71c31425/crates/testing-utils/template_barebones.wasm)
+  which allow you to sign any message which is more than 10 bytes long.
+  - See the [`programs` crate](https://github.com/entropyxyz/programs) for more
+  example programs as well as instructions on how to write and build your own
+  programs.
```
```suggestion
- One or more programs, which define the conditions under which a given message will be signed by
  the Entropy network. The test-cli `register` command takes programs as either the hex-encoded hash
  of an existing program on-chain or the local path to a `.wasm` file containing the compiled
  program.
- The [`device-key-proxy`](https://github.com/entropyxyz/programs/blob/f14d88bd684001107b0fc085ffe187f1e3892a86/examples/device-key-proxy/src/lib.rs)
  program is always available with the zero hash:
  `0000000000000000000000000000000000000000000000000000000000000000`.
- The [`testing-utils`](https://github.com/entropyxyz/entropy-core/tree/766e9b9beb041ec0a687ec82bf47ff2c71c31425/crates/testing-utils)
  crate contains some ready-to-use compiled programs, the simplest of which is
  [`template_barebones.wasm`](https://github.com/entropyxyz/entropy-core/blob/766e9b9beb041ec0a687ec82bf47ff2c71c31425/crates/testing-utils/template_barebones.wasm)
  which allow you to sign any message which is more than 10 bytes long.
- See the [`programs` crate](https://github.com/entropyxyz/programs) for more example programs as
  well as instructions on how to write and build your own programs.
```

HCastano reviewed (approved):
@johnnymatthews yeah those long links can be annoying. I've added some suggestions for how to deal with them.

The default network needs updating, but otherwise looks good!

johnnymatthews commented:
I committed your suggestions @HCastano. Feel free to merge whenever it suits Core.
